### PR TITLE
feat(replays): Transaction Details Replay Section Link

### LINKED ISSUE #39709: Add a button to quickly scroll down to the Replay Preview on the Performance Summary page
At the top of the Issue/Event Details page we've got this new button which achieves two goals:
1. show that there's a replay preview on this page
2. Quickly scroll you down to see that preview

There should be an existing anchor `id="breadcrumbs"` to target.
- only show if the feature is enabled
- only show the button if there is a replay

We need that button added to the Transaction Summary > Event Details page (`/organizations/${org}/performance/${project}:${eventId}/`)

Here might be a good spot, pushed to the right of the space:
Screen Shot 2022-10-05 at 3 00 25 PM

Existing button:
Screen Shot 2022-10-05 at 2 58 11 PM

### PR DESCRIPTION
**Pull request by danecando**
This PR adds a new Replay button to the Transaction Details header. This header component is used by other pages and sections as well. 

### Changes
- Adds a grid column to the EventDetailHeader when a replay is on the page
- Adds Replay button that links to the breadcrumbs section where the replay is located

Closes #39709 

### Tested for style regressions or issues for the following scenarios:

On transaction detail, EventDetailHeaders renders the same on different screen sizes when there is no Replay button added

Screen Shot 2022-10-07 at 11 27 19 AM

Screen Shot 2022-10-07 at 11 27 32 AM

On transaction detail, EventDetailHeaders renders the Replay button just left of the Trace Navigator (sidebar section) on wide screens when a replay is present.

Screen Shot 2022-10-07 at 11 29 15 AM

On transaction detail, EventDetailHeaders renders the replay button in the second row on the right side of the view, with the Trace Navigator to the left taking up 2/3 width for mobile views when a replay is present

Screen Shot 2022-10-07 at 11 31 20 AM

On discover event detail, EventDetailHeaders renders the same when no replay button is added

Screen Shot 2022-10-07 at 11 39 57 AM

Screen Shot 2022-10-07 at 11 40 22 AM

On discover event detail, EventDetailHeaders renders the Replay button just left of the Trace Navigator (sidebar section) on wide screens when a replay is present.

Screen Shot 2022-10-07 at 11 42 23 AM

On discover event detail , EventDetailHeaders renders the replay button in the second row on the right side of the view, with the Trace Navigator to the left taking up 2/3 width for mobile views when a replay is present

Screen Shot 2022-10-07 at 11 42 36 AM

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: static/app/views/performance/transactionDetails/eventMetas.tsx [hunk 1]
```diff
@@ -86,6 +87,10 @@ class EventMetas extends Component<Props, State> {
     } = this.props;
     const {isLargeScreen} = this.state;
 
+    const hasReplay =
+      Boolean(event.entries.find(({type}) => type === 'breadcrumbs')) &&
+      Boolean(event?.tags?.find(({key}) => key === 'replayId')?.value);
```
Comment (ryan953) [inline static/app/views/performance/transactionDetails/eventMetas.tsx:92]: I think we also need to check if the feature is enabled

**Review by ryan953 — APPROVED** [review]

**Comment by danecando** [comment]
@ryan953 Looks like the changes I had to make to the grid are causing a few pixel offset in the visual diff. Not sure if thats a huge deal but I can try to get it pixel perfect if it needs to be

File: static/app/views/performance/transactionDetails/eventMetas.tsx [hunk 2]
```diff
@@ -86,6 +87,13 @@ class EventMetas extends Component<Props, State> {
     } = this.props;
     const {isLargeScreen} = this.state;
 
+    // Replay preview gets rendered as part of the breadcrumb section. We need
+    // to check for precense of both to show the replay link button here.
```
Comment (ryan953) [inline static/app/views/performance/transactionDetails/eventMetas.tsx:91]:
```suggestion
    // to check for presence of both to show the replay link button here.
```

**Comment by ryan953** [comment]
ugh, this might need master pulled in again

**Pull request merged**